requirements-optional.txt

```diff
@@ -0,0 +1,2 @@
+mpi4py
```

Comment:
@ggorman @mloubout probably dask/distributed should end up in this file too, and we should make sure that the relative examples won't run/won't fail if these packages aren't available

Reply:
Obviously unrelated to this PR